README.md

```diff
@@ -1,122 +1,237 @@
-| GH_MAIL | Yes | | The email associated with the token. |
-| GH_NAME | Yes | | The username associated with the token. |
-| EXCLUDED_FILES | No | | The files or directories you want to exclude. Note, we use rsync |
-| REPO | No | `${{ github.repository }}` | The target repository. Default is the current repo. If you specify a different repository (e.g. Andrew-Chen-Wang/github-wiki-action), then you must use a PAT. |
-| WIKI_PUSH_MESSAGE | No | Your commit's message | The message to add to your commit to the wiki git |
-
----
-### Inspiration
-This intended usage was to avoid hosting a private ReadTheDocs
-and instead just use GitHub wiki.
-
-Largely inspired by [wiki-page-creator-action](https://github.com/Decathlon/wiki-page-creator-action)
-and the [issue that arose from it](https://github.com/Decathlon/wiki-page-creator-action/issues/11),
-this GitHub action tries to update the entire wiki based on a single
-directory.
-
----
-### License
+<!--
+FUTURE: Add this paragraph:
 
-```
-   Copyright 2020 Andrew Chen Wang
+⚠️ Make sure that any changes made to the Markdown files in the GitHub Action
+_are also committed_! This GitHub Action uses `git subtree` which takes the
+state of the latest Git commit, not the current state of the working directory.
+You can use [EndBug/add-and-commit] or `git add -A` and
+`git commit --amend --no-edit` to commit changes inside your workflow.
 
-   Licensed under the Apache License, Version 2.0 (the "License");
-   you may not use this file except in compliance with the License.
-   You may obtain a copy of the License at
+-->
 
-       http://www.apache.org/licenses/LICENSE-2.0
+<!--
+FUTURE: Add a second example
 
-   Unless required by applicable law or agreed to in writing, software
-   distributed under the License is distributed on an "AS IS" BASIS,
-   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-   See the License for the specific language governing permissions and
-   limitations under the License.
+```yml
+name: Publish to GitHub wiki
+on:
+  push:
+    branches: [main]
+    paths: [wiki/**, .github/workflows/wiki.yml]
+concurrency:
+  group: wiki
+  cancel-in-progress: true
+permissions:
+  contents: write
+jobs:
+  wiki:
+    environment:
+      name: github-wiki
+      url: ${{ steps.github-wiki-action.outputs.page_url }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - run: date > my-wiki/Todays-date.md
+      - uses: EndBug/add-and-commit@v9
+      - uses: Andrew-Chen-Wang/github-wiki-action@v4
+        id: github-wiki-action
+        with:
+          path: my-wiki
```

Comment:
This doesn't seem like an actual environment variable?

Reply:
The `with: ...` args **do not** need to be environment variables. They _can_ be mapped to environment variables, but you specify env vars in the `env` section like this:

```yml
- uses: jcbhmr/my-action@v1
  with:
    message-template: Hello ${name}!
    to: George Washington
  env:
    NODE_FLAGS=--inspect
```

I note that currently, you use a Docker image which apparently inherits all `with: ...` options as env vars, but that's not always the case. For instance, in @spenserblack's wiki action they use a lowercase and even kebab-case input:

https://github.com/spenserblack/actions-wiki/blob/98f0591597022a21b68f6ea435b34b20e2157ca8/action.yml#L15

Then those inputs can be interpolated into a string that's used by the script:

https://github.com/spenserblack/actions-wiki/blob/98f0591597022a21b68f6ea435b34b20e2157ca8/action.yml#L42

So I can remove this future idea of the renamed re-lowercased params if you like

Reply:
I've updated the actual example since I realized I got ahead of myself and still used `with:` even on the `@v3` version.

So it's fixed now to use `env:`, but we should switch to `with:` in the future. #27